### ISSUE TITLE
Segmentation Fault with Feature set to structural

### ISSUE THREAD

**Issue by gray-sunflower** [jhsk777/SPLASH#1]
Hi, i tried running the code with the command `python main.py --data WIKI --config config/SLIM.yml` but it encounter the segmentation fault issuse in the sampler code. This only happen with --selected_feature set to structural. When i try `python feature_selection.py --data WIKI --config config/SLIM.yml` it also encounter the issuse. When i set different feature it still run smoothly.

**Comment by jhsk777** [comment]
Hello, 
After looking into the issue, it seems that the problem occurs when using 2-hop recent neighbor sampling in TGL. In my environment, the issue does not occur, so I suspect there might be a conflict with updated dependencies. There is a similar issue reported on the TGL GitHub repository, and a proposed solution is available: [https://github.com/amazon-science/tgl/issues/20]



```        
void sample_layer(std::vector<NodeIDType> &_root_nodes, std::vector<TimeStampType> &_root_ts,
                          int neighs, bool use_ptr, bool from_root)
        {
            double t_s = omp_get_wtime();
            std::vector<NodeIDType> *root_nodes;
            std::vector<TimeStampType> *root_ts;
            if (from_root)
            {
                root_nodes = &_root_nodes;
                root_ts = &_root_ts;
            }else { // add a else to initialize them
                root_nodes = &(ret[ret.size() - 1 - num_history].nodes);
                root_ts = &(ret[ret.size() - 1 - num_history].ts);
            }
            double t_ptr_s = omp_get_wtime();
            if (use_ptr) 
                update_ts_ptr(num_history, *root_nodes, *root_ts, 0);
            ret[0].ptr_time += omp_get_wtime() - t_ptr_s;
```
Please try modifying sample_core.cpp as suggested in the link, then run python utils/setup.py build_ext --inplace, and retry your original code. If this resolves the issue, we can proceed to properly update the sample_core.cpp code.

I apologize for the inconvenience, and if you encounter any further problems, please feel free to let me know. Thank you!

**Comment by gray-sunflower** [comment]
Hi, thanks for your suggestion, i have fixed the issuses and have reproduced the main table results. i have one more question regrading the experiments. I really want to reproduce the experiment of Figure 9 (unseen ratio), but i find it hard to understand the settings of the experiments. Is r (unseen ratio) the same as the train/valid ratio. If not, is there any params to set the unseen ratio in the current code to reproduce the results.

**Comment by jhsk777** [comment]
Hello,
The "unseen ratio" refers to the test ratio. As mentioned in the caption of Figure 9 in the paper, the last 10% of the available set is used as the validation set. 
Therefore, if the unseen ratio is 0.4, you can set `train_ratio` to 0.5 and `validation_ratio` to 0.6 in the arguments. This ensures that the test set accounts for 40% of the data, meaning the available data is 60%, and the last 10% of that is used for validation.
If you have any further questions, feel free to ask anytime. Thank you!
